tests/integration/helpers.py

```diff
@@ -51,16 +51,35 @@ def bootstrap_microceph() -> Tuple[CephFSShareInfo, CephFSAuthInfo]:
     _exec_commands(
         instance,
         [
-            ["ln", "-s", "/bin/true", "/usr/local/bin/udevadm"],
-            ["apt-get", "-y", "update"],
-            ["apt-get", "-y", "install", "ceph-common"],
-            ["snap", "install", "microceph"],
-            ["microceph", "cluster", "bootstrap"],
-            ["microceph", "disk", "add", "loop,2G,3"],
+            "ln -s /bin/true".split(),
+            "apt update -y".split(),
+            "apt upgrade -y".split(),
+            "apt install -y ceph-common".split(),
+        ],
+    )
+
+    # Restart in case there was a kernel update.
+    instance.restart(wait=True)
+    _wait_for_microceph(instance)
+
+    _exec_commands(
+        instance,
+        [
+            "snap install microceph".split(),
+            "microceph cluster bootstrap".split(),
+            "microceph disk add loop,2G,3".split(),
```

Comment:
Same as above. Why not use lists directly to save some time?

```suggestion
            ["snap", "install", "microceph"],
            ["microceph", "cluster", "bootstrap"],
            ["microceph", "disk", "add", "loop,2G,3"],
```